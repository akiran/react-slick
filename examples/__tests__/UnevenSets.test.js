import React from "react";
import { mount } from "enzyme";
import UnevenSetsFinite from "../UnevenSetsFinite";
import UnevenSetsInfinite from "../UnevenSetsInfinite";
import { html as beautify_html } from "js-beautify";
import { clickNext } from "../../__tests__/testUtils";

describe("UnevenSets Finite", () => {
  test("Counting test", () => {
    const slider = mount(<UnevenSetsFinite />);
    let totalSlides = slider.find(".slick-slide").length;
    let clonedSlides = slider.find(".slick-cloned").length;
    let activeSlides = slider.find(".slick-slide.slick-active").length;
    let dots = slider.find(".slick-dots").children().length;
    expect(totalSlides).toEqual(6);
    expect(clonedSlides).toEqual(0);
    expect(activeSlides).toEqual(4);
    expect(dots).toEqual(2);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Positioning test", () => {
    const slider = mount(<UnevenSetsFinite />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(0);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2,
      3
    ]);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Activity test", () => {
    const slider = mount(<UnevenSetsFinite />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(0);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2,
      3
    ]);

    clickNext(slider);

    currentSlide = slider.find("div.slick-current");
    activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(4);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      4,
      5
    ]);

    clickNext(slider);

    currentSlide = slider.find("div.slick-current");
    activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(4);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      4,
      5
    ]);

    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
});

describe("UnevenSets Infinite", () => {
  test("Counting test", () => {
    const slider = mount(<UnevenSetsInfinite />);
    let totalSlides = slider.find(".slick-slide").length;
    let clonedSlides = slider.find(".slick-cloned").length;
    let activeSlides = slider.find(".slick-slide.slick-active").length;
    let dots = slider.find(".slick-dots").children().length;
    expect(totalSlides).toEqual(16);
    expect(clonedSlides).toEqual(10);
    expect(activeSlides).toEqual(4);
    expect(dots).toEqual(2);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Positioning test", () => {
    const slider = mount(<UnevenSetsInfinite />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(0);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2,
      3
    ]);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
  test("Activity test", () => {
    const slider = mount(<UnevenSetsInfinite />);
    let currentSlide = slider.find("div.slick-current");
    let activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(0);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      0,
      1,
      2,
      3
    ]);

    clickNext(slider);

    currentSlide = slider.find("div.slick-current");
    activeSlides = slider.find("div.slick-active");
    expect(currentSlide.props()["data-index"]).toEqual(4);
    expect(activeSlides.map(slide => slide.props()["data-index"])).toEqual([
      4,
      5,
      6,
      7
    ]);
    expect(beautify_html(slider.html())).toMatchSnapshot();
  });
});
