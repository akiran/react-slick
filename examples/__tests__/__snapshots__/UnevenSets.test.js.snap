// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnevenSets Finite Activity test 1`] = `
"<div>
    <h2>Uneven sets (finite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"0\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next slick-disabled\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"\\"><button>1</button></li>
            <li class=\\"slick-active\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;

exports[`UnevenSets Finite Counting test 1`] = `
"<div>
    <h2>Uneven sets (finite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev slick-disabled\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"slick-active\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;

exports[`UnevenSets Finite Positioning test 1`] = `
"<div>
    <h2>Uneven sets (finite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev slick-disabled\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"slick-active\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;

exports[`UnevenSets Infinite Activity test 1`] = `
"<div>
    <h2>Uneven sets (infinite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-4\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-3\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-2\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-active slick-cloned\\" aria-hidden=\\"false\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-active slick-cloned\\" aria-hidden=\\"false\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"\\"><button>1</button></li>
            <li class=\\"slick-active\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;

exports[`UnevenSets Infinite Counting test 1`] = `
"<div>
    <h2>Uneven sets (infinite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-4\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-3\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-2\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"slick-active\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;

exports[`UnevenSets Infinite Positioning test 1`] = `
"<div>
    <h2>Uneven sets (infinite)</h2>
    <div class=\\"slick-slider slick-initialized\\" dir=\\"ltr\\"><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-prev\\" style=\\"display: block;\\"> Previous</button>
        <div class=\\"slick-list\\">
            <div class=\\"slick-track\\" style=\\"opacity: 1; transform: translate3d(0px, 0px, 0px);\\">
                <div data-index=\\"-4\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-3\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-2\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"-1\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"0\\" class=\\"slick-slide slick-active slick-current\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"1\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"2\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"3\\" class=\\"slick-slide slick-active\\" tabindex=\\"-1\\" aria-hidden=\\"false\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"4\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"5\\" class=\\"slick-slide\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"outline: none; width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"6\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>1</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"7\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>2</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"8\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>3</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"9\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>4</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"10\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>5</h3>
                        </div>
                    </div>
                </div>
                <div data-index=\\"11\\" tabindex=\\"-1\\" class=\\"slick-slide slick-cloned\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\">
                    <div>
                        <div tabindex=\\"-1\\" style=\\"width: 100%; display: inline-block;\\">
                            <h3>6</h3>
                        </div>
                    </div>
                </div>
            </div>
        </div><button type=\\"button\\" data-role=\\"none\\" class=\\"slick-arrow slick-next\\" style=\\"display: block;\\"> Next</button>
        <ul style=\\"display: block;\\" class=\\"slick-dots\\">
            <li class=\\"slick-active\\"><button>1</button></li>
            <li class=\\"\\"><button>2</button></li>
        </ul>
    </div>
</div>"
`;
